testbed/index.js

```diff
@@ -14,77 +14,86 @@
-        auth.accountName,
-        auth.accountKey
+    return new StorageSharedKeyCredential(auth.accountName, auth.accountKey);
+}
+
+function createAzureContainerClient(auth, containerName) {
+    const sharedKeyCredential = createAzureCredential(auth);
+    const blobServiceClient = new BlobServiceClient(
+        `https://${auth.accountName}.blob.core.windows.net`,
+        sharedKeyCredential
     );
+    return blobServiceClient.getContainerClient(containerName);
+}
+
+function createAzureSAS(auth, containerName, blobName, perm="r") {
+    const containerClient = createAzureContainerClient(auth, containerName);
+
+    const permissions = new BlobSASPermissions();
+    permissions.read = (perm.indexOf("r") >= 0);
+    permissions.write = (perm.indexOf("w") >= 0);
+    permissions.delete = (perm.indexOf("d") >= 0);
+
     const ONE_HOUR_MS = 60 * 60 * 1000;
-    const query = azureStorageBlob.generateBlobSASQueryParameters({
-        protocol: azureStorageBlob.SASProtocol.HTTPS,
-        expiryTime: new Date(Date.now() + ONE_HOUR_MS),
+    const sharedKeyCredential = createAzureCredential(auth);
+    const blobClient = containerClient.getBlockBlobClient(blobName);
+    const query = generateBlobSASQueryParameters({
+        protocol: SASProtocol.Https,
+        expiresOn: new Date(Date.now() + ONE_HOUR_MS),
         containerName,
         blobName,
         permissions
-    }, credential).toString();
-    const path = encodeURIComponent(`${containerName}/${blobName}`);
-    return `https://${auth.accountName}.blob.core.windows.net/${path}?${query}`;
+    }, sharedKeyCredential).toString();
+
+    return `${blobClient.url}?${query}`;
 }
 
 async function commitAzureBlocks(auth, containerName, blobName) {
-    const credential = new azureStorageBlob.SharedKeyCredential(
-        auth.accountName,
-        auth.accountKey
-    );
-    const path = encodeURIComponent(`${containerName}/${blobName}`);
-    const url = `https://${auth.accountName}.blob.core.windows.net/${path}`;
-    const blobURL = new azureStorageBlob.BlockBlobURL(
-        url,
-        azureStorageBlob.BlockBlobURL.newPipeline(credential)
-    );
-    const blockList = await blobURL.getBlockList(
-        azureStorageBlob.Aborter.none,
-        "uncommitted"
-    );
-    await blobURL.commitBlockList(
-        azureStorageBlob.Aborter.none,
-        blockList.uncommittedBlocks.map(x => x.name)
-    );
+    const containerClient = createAzureContainerClient(auth, containerName);
+    const blobClient = containerClient.getBlockBlobClient(blobName);
+    const blockList = await blobClient.getBlockList("uncommitted");
+    await blobClient.commitBlockList(blockList.uncommittedBlocks.map(x => x.name));
 }
 
 async function resolveLocation(value, options) {
-    const url = URL.parse(value);
-    if (url.protocol === "https:" || url.protocol === "http:") {
+    if (value.startsWith("https:") || value.startsWith("http:")) {
```

Comment:
Usually we don't allow `http`. We require `https`...

Reply:
I agree, but this is a test application and I'd like to be able to upload to localhost.